src/app/add-post/add-post.component.ts

```diff
@@ -0,0 +1,40 @@
+import {Component} from '@angular/core';
+import {Post} from "../post.model";
+import {Router} from "@angular/router";
+import {PostService} from "../services/post.service";
+
+@Component({
+  selector: 'app-add-post',
+  templateUrl: './add-post.component.html',
+  styleUrls: ['./add-post.component.scss']
+})
+export class AddPostComponent {
+  post: Post = {title: "", content: "", category: "", likes: 0, publicationDate: ""}
+
+  constructor(private postService: PostService, private router: Router) {
+  }
+
+  async onSubmit() {
+    this.post.publicationDate = new Date().toISOString();
+    this.postService.addPost(this.post).subscribe(
```

Comment:
We're writing `async/await` JavaScript, so stuff like `.subscribe` needs to be avoided.

In the PostService, we should wrap the Observable that gets returned using `lastValueFrom()`. See an example in a comment in the PostService.